File: Python/Hash/longestConsecutive/longestConsecutive.py
class Solution:
    def longestConsecutive(self, nums: list[int]) -> int:
        numSet: set[int] = set(nums)

        maxLen: int = 0

        for num in numSet:
            if num - 1 not in numSet:
                currentNum: int = num
                count: int = 1

                while currentNum + 1 in numSet:
                    currentNum += 1
                    count += 1

                maxLen = max(maxLen, count)

        return maxLen

File: cpp/Hash/longestConsecutive/longestConsecutive.cpp
#include <algorithm>
#include <vector>
#include <unordered_set>
class Solution{
    public:
        int longestConsecutive(std::vector<int> nums){
            std::unordered_set<int> resSet(nums.begin(),nums.end());

            int maxLen = 0;
            for(auto num: resSet){
                if(resSet.find(num-1) == resSet.end()){
                    int  currentNum = num;
                    int count = 1;

                    while(resSet.find(currentNum+1)!=resSet.end()){
                        currentNum++;
                        count++;
                    }

                    maxLen = std::max(count,maxLen);
                }
            }

            return maxLen;
        }
};
